src/proxy.ts

```diff
@@ -49,4 +49,31 @@ function computeApi3ReaderProxyV1Address(
   );
 }
 
-export { computeApi3ReaderProxyV1Address };
+function computeCommunalApi3ReaderProxyV1Address(chainId: ethers.BigNumberish, dapiName: string) {
+  return computeApi3ReaderProxyV1Address(chainId, dapiName, 1, '0x');
+}
+
+function computeDappId(dappAlias: string, chainId: ethers.BigNumberish) {
+  if (!DAPPS.some((dapp) => dapp.alias === dappAlias)) {
+    throw new Error(`dApp with alias ${dappAlias} not registered to the package`);
+  }
+  return ethers.solidityPackedKeccak256(
+    ['bytes32', 'uint256'],
```

Comment:
I changed my mind on this. Chain ID can be a `uint256` and we can assign an arbitrary chain ID to non-EVM chains as necessary. Also, using the hash of the string rather than the string itself makes on-chain derivation of dApp IDs cheaper.